File: sketches/sketches.model/src/test/java/edu/kit/kastel/lissa/sketches/model/TestUMLClass.java
package edu.kit.kastel.lissa.sketches.model;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import edu.kit.kastel.lissa.sketches.model.elements.uml.IUMLClass;
import edu.kit.kastel.lissa.sketches.model.impl.Box;
import edu.kit.kastel.lissa.sketches.model.types.SketchBoxTypes;

public class TestUMLClass {
    private IUMLClass clazz;

    @BeforeEach
    public void setup() {
        this.clazz = SketchBoxTypes.CLASS.map(new Box("TestElement", 0.5), IUMLClass.class);
    }

    @Test
    public void testAddMethod() {
        this.clazz.addMethod("Method");
        assertEquals(1, clazz.getMethods().size());
        assertEquals("Method", clazz.getMethods().get(0));
    }

}

File: sketches/model/src/test/kotlin/edu/kit/kastel/lissa/sketches/model/TestUMLClass.kt
package edu.kit.kastel.lissa.sketches.model

import edu.kit.kastel.lissa.sketches.model.elements.SketchElement
import edu.kit.kastel.lissa.sketches.model.elements.wrapper.ClassNode
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class TestUMLClass : TestBase() {
    private lateinit var clazz: ClassNode

    @BeforeEach
    fun setup() {
        val sketch = Sketch()
        val element = SketchElement()
        element.init(sketch)
        clazz = ClassNode(element)
    }

    @Test
    fun testAddMethod() {
        clazz.addMethod("Method")
        Assertions.assertEquals(1, clazz.methods().size)
        Assertions.assertEquals("Method", clazz.methods()[0])
    }
}
